x-pack/plugins/security_solution/public/common/components/agents/agent_status/endpoint/endpoint_agent_status.tsx

```diff
@@ -0,0 +1,170 @@
+import { getEmptyValue } from '../../../empty_value';
+
+import { useGetEndpointPendingActionsSummary } from '../../../../../management/hooks/response_actions/use_get_endpoint_pending_actions_summary';
+import { useTestIdGenerator } from '../../../../../management/hooks/use_test_id_generator';
+import type { EndpointPendingActions, HostInfo } from '../../../../../../common/endpoint/types';
+import { useGetEndpointDetails } from '../../../../../management/hooks';
+import { getAgentStatusText } from '../../agent_status_text';
+import { AgentResponseActionsStatus } from '../agent_response_action_status';
+
+export const ISOLATING_LABEL = i18n.translate(
+  'xpack.securitySolution.endpoint.agentAndActionsStatus.isIsolating',
+  { defaultMessage: 'Isolating' }
+);
+export const RELEASING_LABEL = i18n.translate(
+  'xpack.securitySolution.endpoint.agentAndActionsStatus.isUnIsolating',
+  { defaultMessage: 'Releasing' }
+);
+export const ISOLATED_LABEL = i18n.translate(
+  'xpack.securitySolution.endpoint.agentAndActionsStatus.isolated',
+  { defaultMessage: 'Isolated' }
+);
+
+const EuiFlexGroupStyled = styled(EuiFlexGroup)`
+  .isolation-status {
+    margin-left: ${({ theme }) => theme.eui.euiSizeS};
+  }
+`;
+
+export interface EndpointAgentStatusProps {
+  endpointHostInfo: HostInfo;
+  /**
+   * If set to `true` (Default), then the endpoint isolation state and response actions count
+   * will be kept up to date by querying the API periodically.
+   * Only used if `pendingActions` is not defined.
+   */
+  autoRefresh?: boolean;
+  /**
+   * The pending actions for the host (as return by the pending actions summary api).
+   * If undefined, then this component will call the API to retrieve that list of pending actions.
+   * NOTE: if this prop is defined, it will invalidate `autoRefresh` prop.
+   */
+  pendingActions?: EndpointPendingActions['pending_actions'];
+  'data-test-subj'?: string;
+}
+
+/**
+ * Displays the status of an Endpoint agent along with its Isolation state or the number of pending
+ * response actions against it.
+ *
+ * TIP: if you only have the Endpoint's `agent.id`, then consider using `EndpointAgentStatusById`,
+ * which will call the needed APIs to get the information necessary to display the status.
+ */
+
+// TODO: used by `EndpointAgentStatusById`
+// remove usage/code when `agentStatusClientEnabled` FF is enabled
+export const EndpointAgentStatus = memo<EndpointAgentStatusProps>(
+  ({ endpointHostInfo, autoRefresh = true, pendingActions, 'data-test-subj': dataTestSubj }) => {
+    const getTestId = useTestIdGenerator(dataTestSubj);
+    const { data: endpointPendingActions } = useGetEndpointPendingActionsSummary(
+      [endpointHostInfo.metadata.agent.id],
+      {
+        refetchInterval: autoRefresh ? DEFAULT_POLL_INTERVAL : false,
+        enabled: !pendingActions,
+      }
+    );
+
+    const [hasPendingActions, hostPendingActions] = useMemo<
```

Comment:
I wonder if this makes sense to split this into two separate useMemos? I find the current logic confusing ;/

Reply:
I think this is alright. It is old code and I believe this was done this way so as to not duplicate logic. I'll leave it as it is unless you feel strongly, in which case I'll suggest you create a tech-debt ticket and we address it later. I want to make only necessary changes to the PR so it is not tedious to review (as all the file moves have made this PR quite large)

Reply:
Alright 👍